### DISCUSSION TITLE
Update manager in Mainsail

### DISCUSSION THREAD

**Discussion by jtenniswood** [xblax/flashforge_ad5m_klipper_mod#76] [General]
This might be a dumb question, but will there be an update manager in mainsail to support upgrades in the future?

**Comment by consp** [comment]
Don't really know for the most part, @xblax might have an idea about it, but klipper would not be updatable, while the "source" has been released it's not possible to create a working build for the mcu so updating beyond v0.11 is not possible at the moment.

KlipperScreen has patches breaking it (e.g. disableling the camera since it crashes it) so that's also not possible, might be doable with a custom repo (like it already is now), My plan is to now-and-then pull the master branch in and merge it.
Mainsail/fluidd: Good question.

Don't know if the updates also install the requirements, but those cannot be updated in situ since it requires cross compiling them.

**Comment by xblax** [comment]
No an automated update manager probably won't / can't happen. With a mod like this the configuration and requirements are too specific. The existing update mangers would not handle this correctly. I also think providing an integrated package of tested components that are known to work together provides the better user-experience.

I guess for some components like Mainsail, Fluidd etc. we will update them via the mod eventually, but that's not a priority right now.
